driver-sync/src/test/functional/com/mongodb/client/ChangeStreamProseTest.java

```diff
@@ -0,0 +1,203 @@
+        try {
+            collection.insertOne(Document.parse("{ x: 1 }"));
+            assertNotNull(cursor.next());
+        } finally {
+            cursor.close();
+        }
+    }
+
+    //
+    // Test that ChangeStream will not attempt to resume on any error encountered while executing an aggregate command.
+    //
+    @Test
+    public void testNoResumeForAggregateErrors() {
+        boolean exceptionFound = false;
+        try {
+            collection.watch(asList(Document.parse("{ $unsupportedStage: { _id : 0 } }"))).iterator();
+        } catch (MongoCommandException e) {
+            exceptionFound = true;
+        }
+        assertTrue(exceptionFound);
+    }
+
+    //
+    // ChangeStream will not attempt to resume after encountering error code 11601 (Interrupted), 136 (CappedPositionLost),
+    // or 237 (CursorKilled) while executing a getMore command.
+    //
+    @Test
+    public void testNoResumeErrors() {
+        assumeTrue(serverVersionAtLeast(4, 0));
+        MongoCursor<ChangeStreamDocument<Document>> cursor = collection.watch().iterator();
+        collection.insertOne(Document.parse("{ x: 1 }"));
+
+        for (int errCode : asList(136, 237, 11601)) {
+            try {
+                setFailPoint("getMore", errCode);
+                cursor.next();
+            } catch (MongoException e) {
+                assertEquals(errCode, e.getCode());
+            }
+        }
+        cursor.close();
+    }
+
+    //
+    // Ensure that a cursor returned from an aggregate command with a cursor id and an initial empty batch
+    // is not closed on the driver side.
+    //
+    @Test
+    public void testCursorNotClosed() {
+        MongoCursor<ChangeStreamDocument<Document>> cursor = collection.watch().iterator();
+        assertNotNull(cursor.getServerCursor());
+    }
+
+    //
+    // Test that the killCursors command sent during the "Resume Process" must not be allowed to throw an exception.
+    //
+    @Test
+    public void testNoExceptionFromKillCursors() {
+        assumeTrue(serverVersionAtLeast(4, 0));
+        MongoCursor<ChangeStreamDocument<Document>> cursor = null;
+
+        try {
+            setFailPoint("killCursors", 184);
+            cursor = collection.watch().iterator();
+            collection.insertOne(Document.parse("{ x: 1 }"));
+            assertNotNull(cursor.next());
+        } finally {
+            if (cursor != null) {
+                cursor.close();
+            }
+        }
+    }
+
+    @Test
+    public void testGetResumeTokenFields() {
+        assumeTrue(serverVersionLessThan(asList(4, 0, 7)));
+
+        MongoCursor<ChangeStreamDocument<Document>> cursor = collection.watch().iterator();
+        collection.insertOne(Document.parse("{ _id: 42, x: 1 }"));
+        collection.insertOne(Document.parse("{ _id: 43, x: 2 }"));
+        BsonDocument startAfterResumeToken = cursor.next().getResumeToken();
+        BsonDocument origResumeToken = cursor.next().getResumeToken();
```

Comment:
nit: change name to `expectedResumeToken`

Reply:
Done